src/Aquifer.Jobs/Subscribers/ResourceContentVersionSimilarityMessageSubscriber.cs

```diff
@@ -0,0 +1,278 @@
+    {
+        return await _dbContext
+                   .ResourceContentVersionSnapshots
+                   .SingleOrDefaultAsync(x => x.Id == snapshotVersionId, ct)
+               ?? throw new InvalidOperationException($"Snapshot version with id {snapshotVersionId} not found");
+    }
+    
+    private async Task<(int, string)> GetMachineTranslationIdAndTextForResourceContentVersion(
+        int resourceContentVersionId,
+        int? machineTranslationId,
+        CancellationToken ct)
+    {
+        var machineTranslations = await _dbContext
+                                        .ResourceContentVersionMachineTranslations
+                                        .Where(x => x.ResourceContentVersionId == resourceContentVersionId)
+                                        .OrderBy(x => x.Id)
+                                        .ToListAsync(ct);
+
+        if (machineTranslations.Count == 0)
+        {
+            throw new InvalidOperationException(
+                $"No Machine translation for ResourceContentVersion with id {resourceContentVersionId} found");
+        }
+
+        if (machineTranslationId is not null && !machineTranslations.Any(m => m.Id != machineTranslationId))
+        {
+            throw new InvalidOperationException(
+                $"Machine translation with id {machineTranslationId} not found for resource content version {resourceContentVersionId}");
+        }
+
+        return GetMachineTranslationIdAndText(machineTranslations, machineTranslationId);
+    }
+    
+    private async Task<(int, string)> GetMachineTranslationTextAndIdForSnapshot(
+        int resourceContentVersionSnapshotId,
+        int? machineTranslationId,
+        CancellationToken ct)
+    {
+        var machineTranslations = await _dbContext
+                                            .ResourceContentVersionMachineTranslations
+                                            .Where(x => x.ResourceContentVersion.ResourceContentVersionSnapshots
+                                                .Any(s => s.Id == resourceContentVersionSnapshotId))
+                                            .OrderBy(x => x.Id)  
+                                            .ToListAsync(ct);
+
+        if (machineTranslations.Count == 0)
+        {
+            throw new InvalidOperationException(
+                $"No Machine translation for Snapshot with id {resourceContentVersionSnapshotId} found");
+        }
+
+        if (machineTranslationId is not null && !machineTranslations.Any(m => m.Id != machineTranslationId))
+        {
+            throw new InvalidOperationException(
+                $"Machine translation with id {machineTranslationId} not found for snapshot version {resourceContentVersionSnapshotId}");
+        }
+        
+        return GetMachineTranslationIdAndText(machineTranslations, machineTranslationId);
+    }
+
+    private (int, string) GetMachineTranslationIdAndText(
```

Comment:
👍 I reviewed this method and it looks like it will handle the FIA steps correctly.